src/lib.rs

```diff
@@ -1689,12 +1690,17 @@ pub fn write_tx_tree(fi: &FrameInvariants, fs: &mut FrameState, cw: &mut Context
         for p in 1..3 {
             let sb_offset = bo.sb_offset().plane_offset(&fs.input.planes[p].cfg);
 
+            let tx_bo = BlockOffset {
+                x: bo.x  - ((bw * tx_size.width_mi() == 1) as usize),
+                y: bo.y  - ((bh * tx_size.height_mi() == 1) as usize)
+            };
+
             let po = PlaneOffset {
                 x: sb_offset.x + partition_x,
                 y: sb_offset.y + partition_y
             };
 
-            encode_tx_block(fi, fs, cw, w, p, &bo, chroma_mode, uv_tx_size, uv_tx_type,
+            encode_tx_block(fi, fs, cw, w, p, &tx_bo, chroma_mode, uv_tx_size, uv_tx_type,
```

Comment:
Ooops... for 4x4 chroma we needed this since the last 4x4 luma position cannot be used for chroma 4x4 ! 
Sorry- my thought didn't reach that far..